### PR TITLE
Allows to put a file even if no command on remote host has been perfo…

### PR DESCRIPTION
**Pull request by fmaupas**
Allows to put a file on remote server despite the fact that no command was sent yet on this server.

### CODE REVIEW

File: jumpssh/session.py [hunk 1]
```diff
@@ -581,6 +581,9 @@ def put(self,
         logger.debug("Copy local file '%s' on remote host '%s' in '%s' as '%s'"
                      % (local_path, self.host, remote_path, self.username))
 
+        # ensure the connection is open
+        #self.open()
```
Comment (t-cas) [inline jumpssh/session.py]: thanks for the fix, this is indeed missing but I would rather directly add the call to `self.open()` in the `SSHSession.file` method as this is the one accessing `paramiko.SFTPClient`

File: tests/test_session.py [hunk 2]
```diff
@@ -500,6 +500,10 @@ def test_put(docker_env):
                                                             port=remotehost_port,
                                                             username='user1',
                                                             password='password1')
+    remotehost_session2 = gateway_session.get_remote_session(host=tests_util.get_host_ip(),
```
Comment (t-cas) [inline tests/test_session.py]: note that doing that, `remotehost_session2` is exactly the same instance than `remotehost_session1` as gateway_session is keeping trace of all existing remote sessions and just return an existing one when matching the input criteria (user, host, port)

Comment (t-cas) [reply tests/test_session.py]: to validate your fix, you can simply close the session before calling `put`

File: jumpssh/session.py (hunk 1 above)
Comment (fmaupas) [reply jumpssh/session.py]: done

File: tests/test_session.py (hunk 2 above)
Comment (fmaupas) [reply tests/test_session.py]: Thanks - I just updated the test.

**Pull request merged**